src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1405,6 +1406,216 @@ impl ToQemuParams for QmpSocket {
     }
 }
 
+// DeviceDriver is the device driver string.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(dead_code)]
+pub enum DeviceDriver {
+    PCIBridge,
+    PCIePCIBridge,
+    RootPort,
+    SwitchUpstreamPort,
+    SwitchDownstreamPort,
+}
```

Comment:
We don't need this enum.  The first two members are never referenced, the rest are exactly once.  Please inline the `pcie-root-port`, `x3130-upstream` and `xio3130-downstream` constants at the respective single point of usage.

Let's not overgeneralise this.